serving/src/main/java/feast/serving/service/RedisBackedJobService.java

```diff
@@ -43,34 +43,40 @@ public RedisBackedJobService(JedisPool jedisPool) {
 
   @Override
   public Optional<Job> get(String id) {
-    Jedis jedis = jedisPool.getResource();
-    String json = jedis.get(id);
-    if (json == null) {
-      return Optional.empty();
-    }
+    Jedis jedis = null;
     Job job = null;
-    Builder builder = Job.newBuilder();
     try {
+      jedis = jedisPool.getResource();
+      String json = jedis.get(id);
+      if (json == null) {
+        return Optional.empty();
+      }
+      Builder builder = Job.newBuilder();
       JsonFormat.parser().merge(json, builder);
       job = builder.build();
     } catch (Exception e) {
       log.error(String.format("Failed to parse JSON for Feast job: %s", e.getMessage()));
```

Comment:
Can we update the error message here? Because the exception can come from failed Redis connection or the Redis value is corrupted and Feast failed to be parse it as a Proto message.

Reply:
Done

Reply:
It's better to use JedisConnectionException as opposed to JedisException, as JedisException also includes JedisDataException, which has nothing to do with connection failure.